### PR TITLE
feat: support Symfony 6

### PR DESCRIPTION
**Pull request by cawolf**

### CODE REVIEW

File: Tests/EventListener/FinishControllerSpanSubscriberTest.php [hunk 1]
```diff
@@ -7,35 +7,33 @@
...
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
 
 class FinishControllerSpanSubscriberTest extends TestCase
 {
     use ProphecyTrait;
 
     private $tracingId;
```
Comment (MollocH) [inline Tests/EventListener/FinishControllerSpanSubscriberTest.php:22]: You could also use typed class properties in here. Also applies for the other tests

File: Tests/EventListener/FinishRootSpanSubscriberTest.php [hunk 2]
```diff
@@ -33,25 +39,23 @@ public function testGetSubscribedEvents(): void
         self::assertArrayHasKey('kernel.finish_request', $this->subject::getSubscribedEvents());
     }
 
-    public function testOnFinishRequestIsMasterRequest(): void
+    public function testOnFinishRequestIsMainRequest(): void
     {
-        $event = $this->prophesize(KernelEvent::class);
-        $event->isMasterRequest()->willReturn(true);
+        $event = new KernelEvent($this->kernel->reveal(), $this->request->reveal(), HttpKernelInterface::MASTER_REQUEST);
```
Comment (MollocH) [inline Tests/EventListener/FinishRootSpanSubscriberTest.php:44]: Maybe add a comment that once Symfony 4.4 is no longer supported, we should switch to HttpKernelInterface::MAIN_REQUEST

File: Tests/EventListener/FinishControllerSpanSubscriberTest.php (hunk 1 above)
Comment (cawolf) [reply Tests/EventListener/FinishControllerSpanSubscriberTest.php:22]: True, done.

File: Tests/EventListener/FinishRootSpanSubscriberTest.php (hunk 2 above)
Comment (cawolf) [reply Tests/EventListener/FinishRootSpanSubscriberTest.php:44]: Good point, added the comment.

**Review by MollocH — APPROVED** [review]
LGTM

**Comment by cawolf** [comment]
relates to #75

**Pull request merged**